### PR TITLE
CA Context

### PR DESCRIPTION
**Pull request by norisalsaadie**
components/ca-context.js -> migrated from old to new.
examples/ca-context.html -> migrated from old to new.
test/ca-context-test.html -> test place holder.

### CODE REVIEW

File: components/ca-context.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            return 'hidden';
+        }
+
+        /**
+         * Attempts to construct a path based on the users UI
+         * We use query selector over $ so we only get the first instance of any object
+         * @returns {string} path based values
+         */
+        getUserPath() {
+            const txtProp = ('textContent' in document.createElement('i')) ? 'textContent' : 'innerText';
```
Comment (tapina) [inline components/ca-context.js]: Why not in the module declaration closure at the top like the original?

Comment (norisalsaadie) [inline components/ca-context.js]: A terrible reason - in the original, the function is only invoked on the .value getter. This functionality isn't used anywhere else, so if anything, it should either move into the getter or become part of the class. It doesn't feel very much like a private function to me. Happy to roll it back, though.

Comment (tapina) [inline components/ca-context.js]: txtProp. But as you said on the other PR, it's IE8 so we can remove it asap.

File: components/ca-context.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     * Web component context
+     */
+    class Context extends HTMLElement {
+
+        /** @returns {string} attribute id */
+        get id() {
+            return this.getAttribute('id');
+        }
+
+        /** @param {string} value, assigned to id attrinute */
```
Comment (joeburton) [inline components/ca-context.js]: nit picking especially coming from me with my funky spelling, but attribute is spelt wrong.

File: components/ca-context.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        set id(value) {
+            this.setAttribute('id', value);
+        }
+
+        /** @returns {string} attribute name */
+        get name() {
+            return this.getAttribute('name');
+        }
+
+        /** @param {string} value, assigned to name attrinute */
```
Comment (joeburton) [inline components/ca-context.js]: nit picking especially coming from me with my funky spelling but attribute is spelt wrong.

File: components/ca-context.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        /**
+         * Attempts to construct a path based on the users UI
+         * We use query selector over $ so we only get the first instance of any object
+         * @returns {string} path based values
+         */
+        getUserPath() {
+            const txtProp = ('textContent' in document.createElement('i')) ? 'textContent' : 'innerText';
+            const result = [];
+
+            const nav = document.querySelector('ca-nav:not(.non-contextual) a[data-selected]');
```
Comment (joeburton) [inline components/ca-context.js]: not sure about levels of browser support for :not() caniuse is a bit confusing.
http://caniuse.com/#search=%3Anot

Comment (norisalsaadie) [inline components/ca-context.js]: Good point, but as it's being migrated this is secondary stage.

**Comment by norisalsaadie** [comment]
@joeburton there was no example in the original. The task at hand at this stage is migrating, and todo placeholders for tests.

**Comment by joeburton** [comment]
Approved with the caveat that I really need to see the new component/ file working in context to a project or a decent example which this is not :(

**Pull request merged**